var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -0,0 +1,236 @@
+            python_exec = spec['python'].command.path
+
+            options.append('--with-nrnpython=%s' % python_exec)
+            options.append('--disable-pysetup')
+
+            if spec.satisfies('+cross-compile'):
+                # on cross compile builds we have to provide
+                # PYINCDIR, PYLIB, PYLIBDIR etc
+
+                py_lib = spec['python'].prefix.lib
+                py_inc = spec['python'].headers.directories[0]
+
+                if not os.path.isdir(py_lib):
+                    py_lib = spec['python'].prefix.lib64
+
+                options.extend(['PYINCDIR=%s' % (py_inc),
+                                'PYLIB=-L%s -l%s' % (py_lib, py_version),
+                                'PYLIBDIR=%s' % py_lib,
+                                'PYLIBLINK=-L%s -l%s' % (py_lib, py_version)])
+        else:
+            options.append('--without-nrnpython')
+        return options
+
+    def get_compiler_options(self, spec):
+        options = []
+        flags = self.get_opt_flags()
+
+        # TODO: issue with static build and pgi compiler.
+        # need to add fpic and enabled-shared
+        if spec.satisfies('%pgi'):
+            flags += ' ' + self.compiler.pic_flag
+            options.extend(['CFLAGS=%s' % flags,
+                            'CXXFLAGS=%s' % flags,
+                            '--enable-shared'])
+        else:
+            options.extend(['CFLAGS=%s' % flags,
+                            'CXXFLAGS=%s' % flags])
+
+        return options
+
+    def get_configure_options(self, spec):
+        options = []
+
+        if spec.satisfies('+static'):
+            options.extend(['--disable-shared',
+                            'linux_nrnmech=no'])
+
+        # on os-x disable building carbon 'click' utility (deprecated)
+        if (sys.platform == 'darwin'):
+            options.append('macdarwin=no')
+
+        options.extend(self.get_arch_options(spec))
+        options.extend(self.get_python_options(spec))
+        options.extend(self.get_compiler_options(spec))
+        return options
+
+    def build_nmodl(self, spec, prefix):
+        # TODO: NEURON has two stage compilation for systems
+        # like cray and bg-q. On these platforms it's ok to
+        # use gcc and g++ as front-end compilers.
+        options = ['--prefix=%s' % prefix,
+                   '--with-nmodl-only',
+                   '--without-x',
+                   'CC=%s' % which("gcc"),
+                   'CXX=%s' % which("g++")]
+
+        configure = Executable(join_path(self.stage.source_path, 'configure'))
+        configure(*options)
+        make()
+        make('install')
+
+    def install(self, spec, prefix):
+        c_compiler = spack_cc
+        cxx_compiler = spack_cxx
+
+        # for bg-q we can't set XL as CC and CXX compiler
+        if 'bgq' in self.spec.architecture:
+            c_compiler = spec['mpi'].mpicc
+            cxx_compiler = spec['mpi'].mpicxx
```

Comment:
This isn't going to work if someone tries to build `neuron~mpi`. You need to check whether or not `'+mpi' in spec`.